[RFC] treewide: validate unified uImage.FIT images before flashing

dangowrt opened the pull request:
Prevent flashing truncated or otherwise corrupted uImage.FIT images by verifying checksums and hashes of all sub-images before flashing using the newly packaged `fit_check_sign` tool.

Requires https://github.com/openwrt/openwrt/pull/18370 to be merged first.

dangowrt committed:
treewide: validate unified uImage.FIT images before flashing

Prevent flashing truncated or otherwise corrupted uImage.FIT images
by verifying checksums and hashes of all sub-images before flashing
using the newly packaged fit_check_sign tool.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>

dangowrt committed:
base-files: allow platform_check_image to prevent --force

Introduce special handling for return code 74 (EBADMSG) of
platform_check_image which will mark the image as broken and hence
not allow the user to override the check using the --force option.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>

KanjiMonster commented on target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh:
```diff
@@ -187,22 +187,40 @@ PART_NAME=firmware
-			return 1
-		}
-		return 0
+	cmcc,a10-ubootmod|\
+	cmcc,rax3000m|\
+	gatonetworks,gdsp|\
+	h3c,magic-nx30-pro|\
+	jcg,q30-pro|\
+	jdcloud,re-cp-03|\
+	mediatek,mt7981-rfb|\
+	mediatek,mt7988a-rfb|\
+	mercusys,mr90x-v1-ubi|\
+	nokia,ea0326gmp|\
+	openwrt,one|\
+	netcore,n60|\
+	qihoo,360t7|\
+	routerich,ax3000-ubootmod|\
+	tplink,tl-xdr4288|\
+	tplink,tl-xdr6086|\
+	tplink,tl-xdr6088|\
+	tplink,tl-xtr8488|\
+	xiaomi,mi-router-ax3000t-ubootmod|\
+	xiaomi,redmi-router-ax6000-ubootmod|\
+	xiaomi,mi-router-wr30u-ubootmod|\
+	zyxel,ex5601-t0-ubootmod)
```
Where do all these platforms come from? They weren't there before, and this change isn't explained in the commit message.

dangowrt replied:
Most boards using uImage.FIT were previously not listed and hence the check() function was too broad. I'll update the commit message accordingly.

dangowrt replied:
@KanjiMonster I made the addition of the missing boards to `platform_check_image` an additional commit. I believe in that way it's more clean.

dangowrt committed:
mediatek: filogic: add missing boards to platform_check_image

Add all boards using uImage.FIT to platform_check_image() case section
making sure we are dealing with a FIT image.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>

rmilecki commented:
Nice `validate_firmware_image` change!
